CMakeLists.txt

```diff
@@ -1,168 +1,186 @@
-  install (FILES ${CMAKE_CURRENT_BINARY_DIR}/libcmaes.pc DESTINATION ${RELATIVE_INSTALL_LIB_DIR}/pkgconfig)
-endif ()
-
-
-enable_testing ()
-
-add_subdirectory (src)
-if (BUILD_EXAMPLES)
-  add_subdirectory (examples)
-endif ()
-if (BUILD_TESTS)
-  add_subdirectory (tests)
-endif ()
-
-if (Boost_FOUND AND NUMPY_FOUND)
-  add_subdirectory (python)
-endif ()
+cmake_minimum_required (VERSION 2.8)
+
+if (NOT DEFINED CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE Release CACHE STRING "Build type")
+endif ()
+
+project (libcmaes)
+
+list (APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
+
+
+option (BUILD_SHARED_LIBS "Build libcmaes as a shared library" ON)
+option (BUILD_PYTHON "build python bindings" ON)
+option (BUILD_TESTS "build tests" ON)
+option (BUILD_EXAMPLES "build examples" ON)
+option (USE_OPENMP "Use OpenMP for multithreading" ON)
+option (USE_TBB "Use TBB for multithreading" OFF)
+option (ENABLE_SURROG "support for surrogates" ON)
+set (HAVE_SURROG ${ENABLE_SURROG})
+option (USE_COMPILE_FEATURES "use cmake>=3.1 cxx11 detection" ON)
+option (LINK_PYTHON "link python libraries" ON)
+
+# Offer the user the choice of overriding the installation directories
+set (INSTALL_LIB_DIR     lib${LIB_SUFFIX} CACHE PATH "Installation directory for libraries")
+set (INSTALL_BIN_DIR     bin              CACHE PATH "Installation directory for executables")
+set (INSTALL_INCLUDE_DIR include          CACHE PATH "Installation directory for header files")
+set (INSTALL_DATA_DIR    share/libcmaes       CACHE PATH "Installation directory for data files")
+set (INSTALL_CMAKE_DIR   ${INSTALL_LIB_DIR}/cmake/libcmaes CACHE PATH "Installation directory for cmake config files")
+
+# Make relative paths absolute (needed later on)
+foreach (p LIB BIN INCLUDE DATA CMAKE)
+  set (var INSTALL_${p}_DIR)
+  set (RELATIVE_INSTALL_${p}_DIR ${INSTALL_${p}_DIR})
+  if (NOT IS_ABSOLUTE "${${var}}")
+    set(${var} "${CMAKE_INSTALL_PREFIX}/${${var}}")
+  endif ()
+endforeach ()
+
+
+set (CMAKE_INSTALL_RPATH ${INSTALL_LIB_DIR})
+set (CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+if (POLICY CMP0042)
+  # Set MACOSX_RPATH to ON
+  cmake_policy (SET CMP0042 NEW)
+endif()
+
+include (CheckIncludeFile)
+include (CheckCXXCompilerFlag)
+
+check_include_file (dlfcn.h HAVE_DLFCN_H)
+check_include_file (unistd.h HAVE_UNISTD_H)
+check_include_file (string.h HAVE_STRING_H)
+check_include_file (strings.h HAVE_STRINGS_H)
+check_include_file (inttypes.h HAVE_INTTYPES_H)
+check_include_file (memory.h HAVE_MEMORY_H)
+check_include_file (stdlib.h HAVE_STDLIB_H)
+check_include_file (stdint.h HAVE_STDINT_H)
+check_include_file (sys/types.h HAVE_SYS_TYPES_H)
+check_include_file (sys/stat.h HAVE_SYS_STAT_H)
+
+# eigen
+find_package (Eigen3 NO_MODULE QUIET)
+if (EIGEN3_FOUND)
+  include (${EIGEN3_USE_FILE})
+  message(STATUS "Eigen3 found: ${EIGEN3_INCLUDE_DIRS}")
+else ()
+  find_package (Eigen3 REQUIRED)
+  include_directories (${EIGEN3_INCLUDE_DIRS})
+endif()
+
+if (USE_OPENMP AND USE_TBB)

```

Comment:
USE_TBB could be on, but not checked if omp is already available, or hte other way around.

Reply:
> USE_TBB could be on, but not checked if omp is already available, or hte other way around.

This should be fixed now..